src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -172,86 +172,123 @@ def handle_sso_command(cmd):
-        else:
-            sp_x_509_cert = ''
-            sp_private_key = ''
-
-        if os.path.isfile(idp_metadata):
-            warnings.warn(
-                "Please prepend 'file://' to indicate a local SAML2 IdP file", DeprecationWarning)
-            with open(idp_metadata, 'r', encoding='utf-8') as f:
-                idp_settings = Saml2Parser.parse(f.read(), entity_id=idp_entity_id)
-        elif parse.urlparse(idp_metadata)[0] in ('http', 'https', 'file'):
-            idp_settings = Saml2Parser.parse_remote(
-                url=idp_metadata, validate_cert=False, entity_id=idp_entity_id)
-        else:
-            idp_settings = Saml2Parser.parse(idp_metadata, entity_id=idp_entity_id)
-
-        url_prefix = prepare_url_prefix(mgr.get_module_option('url_prefix', default=''))
-        settings = {
-            'sp': {
-                'entityId': '{}{}/auth/saml2/metadata'.format(ceph_dashboard_base_url, url_prefix),
-                'assertionConsumerService': {
-                    'url': '{}{}/auth/saml2'.format(ceph_dashboard_base_url, url_prefix),
-                    'binding': "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST"
-                },
-                'attributeConsumingService': {
-                    'serviceName': "Ceph Dashboard",
-                    "serviceDescription": "Ceph Dashboard Service",
-                    "requestedAttributes": [
-                        {
-                            "name": idp_username_attribute,
-                            "isRequired": True
-                        }
-                    ]
-                },
-                'singleLogoutService': {
-                    'url': '{}{}/auth/saml2/logout'.format(ceph_dashboard_base_url, url_prefix),
-                    'binding': 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
-                },
-                "x509cert": sp_x_509_cert,
-                "privateKey": sp_private_key
-            },
-            'security': {
-                "nameIdEncrypted": has_sp_cert,
-                "authnRequestsSigned": has_sp_cert,
-                "logoutRequestSigned": has_sp_cert,
-                "logoutResponseSigned": has_sp_cert,
-                "signMetadata": has_sp_cert,
-                "wantMessagesSigned": has_sp_cert,
-                "wantAssertionsSigned": has_sp_cert,
-                "wantAssertionsEncrypted": has_sp_cert,
-                "wantNameIdEncrypted": False,  # Not all Identity Providers support this.
-                "metadataValidUntil": '',
-                "wantAttributeStatement": False
-            }
-        }
-        settings = Saml2Parser.merge_settings(settings, idp_settings)
-        mgr.SSO_DB.saml2.onelogin_settings = settings
-        mgr.SSO_DB.protocol = 'saml2'
-        mgr.SSO_DB.save()
-        return 0, json.dumps(mgr.SSO_DB.saml2.onelogin_settings), ''
+        ret = _handle_saml_setup(cmd)
+        return ret
```

Comment:
For example here (and no need of `if cmd['prefix']` stuff):
```py
@CLICommand('dashboard sso setup saml2')
def _handle_saml_setup(self, cmd: str):
....
```
That automatically 'generates' the prefix handling & callback.